receiver/haproxyreceiver/factory.go

```diff
@@ -0,0 +1,81 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package haproxyreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/haproxyreceiver/internal/metadata"
+)
+
+const (
+	typeStr   = "haproxy"
+	stability = component.StabilityLevelDevelopment
+)
+
+// NewFactory creates a new HAProxy receiver factory.
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		CreateDefaultConfig,
+		receiver.WithMetrics(createReceiverFunc(), stability))
+}
+
+func CreateDefaultConfig() component.Config {
```

Comment:
In order to comply to the [naming conventions](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#naming-convention) of the project, this should be 

```suggestion
func newDefaultConfig() component.Config {
```

Since it doesn't need to be exported and it is a new instance of config.

Reply:
It's exported to run in the components tests.

Reply:
While this is technically safe to export, it is recommended that this be called via the `NewFactory().CreateDefaultConfig()` (forget the exact name) to keep external access consistent across each component.

So you can do that as part of the tests or drop the `_test` as part of the package name and access the unexported methods.